syntax = "proto3";

package product_read;

import "product.proto";

option java_package = "com.community.catalog.proto.product_read";
option java_multiple_files = true;
option go_package = "github.com/community-platform/product_read;product_read";

service ProductReadService {
  rpc GetProductById (GetProductByIdRequest) returns (product.Product);
  rpc GetAllProducts (GetAllProductsRequest) returns (GetAllProductsResponse);
  rpc SearchProducts(SearchProductsRequest) returns (SearchProductsResponse);
  // Retrieves details for a specific product
  rpc GetProductDetails(GetProductDetailsRequest) returns (product.Product);
}

message GetProductByIdRequest {
  string id = 1;
}

message GetAllProductsRequest {}

message GetAllProductsResponse {
  repeated product.Product products = 1;
}

message SearchProductsRequest {
  string query = 1;
}

message SearchProductsResponse {
  repeated product.Product products = 1;
}

// Request to get details for a product
message GetProductDetailsRequest {
  string product_id = 1;
}
